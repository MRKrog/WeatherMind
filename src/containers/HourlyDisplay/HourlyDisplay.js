import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { HourDetail } from '../../components/HourDetail/HourDetail';

export class HourlyDisplay extends Component {
  render() {
    const { hourly } = this.props
    return (
      <div className="HourlyDisplay">
        {
          hourly.map((hour, index) => {
            return <HourDetail key={hour.time} {...hour} index={index}/>
          })
        }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  hourly: state.hourly
})

HourlyDisplay.propTypes = {
  hourly: PropTypes.array.isRequired,
}

export default connect(mapStateToProps)(HourlyDisplay)
