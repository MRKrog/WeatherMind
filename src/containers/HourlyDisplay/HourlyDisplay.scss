.HourlyDisplay {
  display: flex;
  width: 100%;
  padding: 1em;
  overflow-x: scroll;
  background: #1d162b;
  color: #fff;
  border-top: #cccccc solid 1px;
  border-bottom: #cccccc solid 1px;
}
