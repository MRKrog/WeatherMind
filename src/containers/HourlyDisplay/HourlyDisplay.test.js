import React, { Component } from 'react';
import { HourlyDisplay, mapStateToProps } from './HourlyDisplay';
import { shallow } from 'enzyme';

const mockHourlyData = [
  {time: "20", icon: "partly-cloudy-night", temperature: 63},
  {time: "21", icon: "cloudy", temperature: 60},
  {time: "22", icon: "partly-cloudy-night", temperature: 56}
]

describe('HourlyDisplay', () => {

  describe('HourDisplay Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<HourlyDisplay hourly={mockHourlyData} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

  })


  describe('mapStateToProps', () => {

    it('should return an object with the hourly array of weather', () => {})
      const mockState = {
        hourly: [
          {time: "20", icon: "partly-cloudy-night", temperature: 63},
          {time: "21", icon: "cloudy", temperature: 60},
          {time: "22", icon: "partly-cloudy-night", temperature: 56}
        ],
        filter: "SET_HOURLY"
      }
      const expected = {
        hourly: [
          {time: "20", icon: "partly-cloudy-night", temperature: 63},
          {time: "21", icon: "cloudy", temperature: 60},
          {time: "22", icon: "partly-cloudy-night", temperature: 56}
        ],
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
  })

})
