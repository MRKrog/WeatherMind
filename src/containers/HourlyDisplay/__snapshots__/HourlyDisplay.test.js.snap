// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyDisplay HourDisplay Component should match the snapshot 1`] = `
<div
  className="HourlyDisplay"
>
  <HourDetail
    icon="partly-cloudy-night"
    index={0}
    key="20"
    temperature={63}
    time="20"
  />
  <HourDetail
    icon="cloudy"
    index={1}
    key="21"
    temperature={60}
    time="21"
  />
  <HourDetail
    icon="partly-cloudy-night"
    index={2}
    key="22"
    temperature={56}
    time="22"
  />
</div>
`;
